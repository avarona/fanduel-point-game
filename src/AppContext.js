import React from 'react';

export const AppContext = React.createContext();

export const AppContextProvider = ({ value, children }) => {
    return (
        <AppContext.Provider value={value}>
            {children}
        </AppContext.Provider>
    )
}
