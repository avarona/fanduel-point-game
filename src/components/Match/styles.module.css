.matchContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
    max-width: 800px;
}

.players {
    display: flex;
    align-items: center;
    justify-content: space-around;
    margin-bottom: 48px;
}

.textBetween {
    font-size: 42px;
}

.nextButton, .endButton {
    align-self: center;
    width: 200px;
    padding: 12px;
    border-radius: 4px;
    cursor: pointer;
}
.nextButton {
    background: cyan;
}

.endButton {
    margin-top: 12px;
    background: red;
    color: white;
}

.endGame {
    display: flex;
    flex-direction: column;
    align-self: center;
}


@media only screen and (max-width: 450px) {
    .players {
        flex-direction: column;
    }
}
