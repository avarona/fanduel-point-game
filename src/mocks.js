export const players = [
    {id: 1, first_name: 'Kobe', last_name: 'Bryant', fppg: 15, images: { default: { url: 'slick pic' }}},
    {id: 2, first_name: 'Lebron', last_name: 'James', fppg: 10, images: { default: { url: 'the dunker' }}},
    {id: 3, first_name: 'Michael', last_name: 'Jordan', fppg: 5, images: { default: { url: 'the legend' } }}
];

export const matchPlayers = [players[0], players[1]];

export const playerDetails = players[0];