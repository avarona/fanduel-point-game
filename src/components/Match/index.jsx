import React from 'react';
import Player from '../Player';
import styles from './styles.module.css';
import { AppContext } from '../../AppContext';

const Match = ({ players }) => {
    const [player1, player2] = players;
    return (
        <AppContext.Consumer>
            {({ winnerId, nextMatch, totalGuessCount, correctGuessCount, resetGame }) => (
                <div className={styles.matchContainer}>
                    <div className={styles.guessCounter}>
                        <span className={styles.guessTitle}>Win / Loss</span>
                        <span>{correctGuessCount} / {totalGuessCount - correctGuessCount}</span>
                    </div>

                    <div className={styles.players}>
                        <Player details={player1} />
                        <span className={styles.textBetween}>VS</span>
                        <Player details={player2} />
                    </div>

                    {/* show next button only when winner is calculated */}
                    {winnerId && 
                        (correctGuessCount < 10
                            ? <button className={styles.nextButton} onClick={nextMatch}>NEXT</button>
                            : <div className={styles.endGame}>
                                <span className={styles.endText}>Congrats, you made 10 correct guesses!</span>
                                <button className={styles.endButton} onClick={resetGame}>RESET</button>
                            </div>)
                    }
                </div>
            )}
        </AppContext.Consumer>
    )
};

export default Match;