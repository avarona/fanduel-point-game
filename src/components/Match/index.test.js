import React from 'react';
import { mount } from 'enzyme';
import Match from './';
import * as Mocks from '../../mocks';
import { AppContextProvider } from '../../AppContext';

describe('Match Component', () => {
    describe('renders', () => {
        it('with default props', () => {
            const context = { winnerId: undefined };
            const wrapper = mount(
                <AppContextProvider value={context}>
                    <Match players={Mocks.matchPlayers} />
                </AppContextProvider>
            );

            expect(wrapper.html()).toMatchSnapshot();
        });
    
        it('the `next` button when game is played', () => {
            const context = { winnerId: undefined };
            const wrapper = mount(
                <AppContextProvider value={context}>
                    <Match players={Mocks.matchPlayers} />
                </AppContextProvider>
            );

            expect(wrapper.html()).toMatchSnapshot();
        });
    
        it('the end game status when game finishes', () => {
            const context = { winnerId: 1, correctGuessCount: 10 };
            const wrapper = mount(
                <AppContextProvider value={context}>
                    <Match players={Mocks.matchPlayers} />
                </AppContextProvider>
            );

            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
