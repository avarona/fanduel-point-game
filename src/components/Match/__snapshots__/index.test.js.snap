// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Component renders the \`next\` button when game is played 1`] = `"<div class=\\"matchContainer\\"><div class=\\"players\\"><div class=\\"playerContainer\\"><img class=\\"image\\" src=\\"slick pic\\" alt=\\"Kobe Bryant\\"><span class=\\"name\\">Kobe Bryant</span></div><span class=\\"textBetween\\">VS</span><div class=\\"playerContainer\\"><img class=\\"image\\" src=\\"the dunker\\" alt=\\"Lebron James\\"><span class=\\"name\\">Lebron James</span></div></div></div>"`;

exports[`Match Component renders the end game status when game finishes 1`] = `"<div class=\\"matchContainer\\"><div class=\\"players\\"><div class=\\"playerContainer disabled\\"><img class=\\"image winnerSelected\\" src=\\"slick pic\\" alt=\\"Kobe Bryant\\"><span class=\\"name\\">Kobe Bryant</span><span class=\\"score\\">15.00</span><img class=\\"winnerIcon\\" alt=\\"check mark\\" src=\\"https://cdn.pixabay.com/photo/2017/01/13/01/22/ok-1976099_960_720.png\\"></div><span class=\\"textBetween\\">VS</span><div class=\\"playerContainer disabled\\"><img class=\\"image winnerSelected\\" src=\\"the dunker\\" alt=\\"Lebron James\\"><span class=\\"name\\">Lebron James</span><span class=\\"score\\">10.00</span><img class=\\"winnerIcon\\" alt=\\"x mark\\" src=\\"https://cdn.pixabay.com/photo/2017/02/12/21/29/false-2061131_960_720.png\\"></div></div><div class=\\"endGame\\"><span class=\\"endText\\">Congrats, you made 10 correct guesses!</span><button class=\\"endButton\\">RESET</button></div></div>"`;

exports[`Match Component renders with default props 1`] = `"<div class=\\"matchContainer\\"><div class=\\"players\\"><div class=\\"playerContainer\\"><img class=\\"image\\" src=\\"slick pic\\" alt=\\"Kobe Bryant\\"><span class=\\"name\\">Kobe Bryant</span></div><span class=\\"textBetween\\">VS</span><div class=\\"playerContainer\\"><img class=\\"image\\" src=\\"the dunker\\" alt=\\"Lebron James\\"><span class=\\"name\\">Lebron James</span></div></div></div>"`;
